{"name": "Залогиненный пользователь может оформить заказ", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: //section[contains(@class, \"Modal_modal_opened\")]/div[contains(@class, \"Modal_modal__container\")]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16", "trace": "self = <test_main_page.TestMainPage object at 0x1094dc950>, driver = <selenium.webdriver.firefox.webdriver.WebDriver (session=\"18ddc109-4c6a-4a2d-a64d-2da384589046\")>\nset_user_tokens = None\n\n    @allure.title('Залогиненный пользователь может оформить заказ')\n    def test_making_order_by_authenticated_user_success(self, driver, set_user_tokens):\n        main_page = MainPage(driver)\n        main_page.click_on_button_login_in_main()\n        main_page.drag_and_drop_ingredient_to_order()\n        main_page.click_on_button_make_order()\n>       assert main_page.check_displaying_of_confirmation_modal_of_order()\n\ntests/test_main_page.py:47: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\npage_objects/main_page.py:72: in check_displaying_of_confirmation_modal_of_order\n    return self.check_displaying_of_element(MainPageLocators.confirmation_modal_of_order)\npage_objects/base_page.py:41: in check_displaying_of_element\n    return self.driver.find_element(*locator).is_displayed()\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py:741: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py:347: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x10952ec30>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"Unable to locate element: //section[contains(.../content/shared/webdriver/Errors.sys.mjs:511:5\\\\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: //section[contains(@class, \"Modal_modal_opened\")]/div[contains(@class, \"Modal_modal__container\")]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nE       WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5\nE       NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5\nE       dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\n\n/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py:229: NoSuchElementException"}, "steps": [{"name": "Кликнуть по кнопке \"Войти в аккаунт\" на главной", "status": "passed", "steps": [{"name": "Кликнуть на элемент", "status": "passed", "steps": [{"name": "Проверить кликабельность элемента", "status": "passed", "parameters": [{"name": "locator", "value": "('xpath', './/button[text() = \"Войти в аккаунт\"]')"}], "start": 1726477692725, "stop": 1726477692740}], "parameters": [{"name": "locator", "value": "('xpath', './/button[text() = \"Войти в аккаунт\"]')"}], "start": 1726477692725, "stop": 1726477693014}], "start": 1726477692725, "stop": 1726477693014}, {"name": "Добавить интгридиенты", "status": "passed", "steps": [{"name": "Найти элемент на странице", "status": "passed", "steps": [{"name": "Подождать прогрузки элемента", "status": "passed", "parameters": [{"name": "locator", "value": "('xpath', './/*[@alt=\"Флюоресцентная булка R2-D3\"]')"}], "start": 1726477693014, "stop": 1726477693023}], "parameters": [{"name": "locator", "value": "('xpath', './/*[@alt=\"Флюоресцентная булка R2-D3\"]')"}], "start": 1726477693014, "stop": 1726477693024}, {"name": "Найти элемент на странице", "status": "passed", "steps": [{"name": "Подождать прогрузки элемента", "status": "passed", "parameters": [{"name": "locator", "value": "('xpath', '//section[contains(@class, \"BurgerConstructor_basket\")]')"}], "start": 1726477693025, "stop": 1726477693031}], "parameters": [{"name": "locator", "value": "('xpath', '//section[contains(@class, \"BurgerConstructor_basket\")]')"}], "start": 1726477693025, "stop": 1726477693033}, {"name": "Перетащить элемент", "status": "passed", "parameters": [{"name": "source_element", "value": "<selenium.webdriver.remote.webelement.WebElement (session=\"18ddc109-4c6a-4a2d-a64d-2da384589046\", element=\"2888259e-4ed4-4707-94b9-340718be4bd1\")>"}, {"name": "target_element", "value": "<selenium.webdriver.remote.webelement.WebElement (session=\"18ddc109-4c6a-4a2d-a64d-2da384589046\", element=\"2e6721a5-7bc9-4b02-8668-91dbbac562ef\")>"}], "start": 1726477693033, "stop": 1726477698608}], "start": 1726477693014, "stop": 1726477698608}, {"name": "Кликнуть на кнопку создания заказа", "status": "passed", "steps": [{"name": "Кликнуть на элемент", "status": "passed", "steps": [{"name": "Проверить кликабельность элемента", "status": "passed", "parameters": [{"name": "locator", "value": "('class name', 'button_button__33qZ0')"}], "start": 1726477698608, "stop": 1726477698737}], "parameters": [{"name": "locator", "value": "('class name', 'button_button__33qZ0')"}], "start": 1726477698608, "stop": 1726477699021}], "start": 1726477698608, "stop": 1726477699021}, {"name": "Проверить отображение окна о создании заказа", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: //section[contains(@class, \"Modal_modal_opened\")]/div[contains(@class, \"Modal_modal__container\")]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\n\n", "trace": "  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/kosova.ea/Diplom/Diplom_3/page_objects/main_page.py\", line 72, in check_displaying_of_confirmation_modal_of_order\n    return self.check_displaying_of_element(MainPageLocators.confirmation_modal_of_order)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/kosova.ea/Diplom/Diplom_3/page_objects/base_page.py\", line 41, in check_displaying_of_element\n    return self.driver.find_element(*locator).is_displayed()\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 741, in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "steps": [{"name": "Проверить отображение элемента", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: //section[contains(@class, \"Modal_modal_opened\")]/div[contains(@class, \"Modal_modal__container\")]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\n\n", "trace": "  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/kosova.ea/Diplom/Diplom_3/page_objects/base_page.py\", line 41, in check_displaying_of_element\n    return self.driver.find_element(*locator).is_displayed()\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 741, in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "parameters": [{"name": "locator", "value": "('xpath', '//section[contains(@class, \"Modal_modal_opened\")]/div[contains(@class, \"Modal_modal__container\")]')"}], "start": 1726477699021, "stop": 1726477699023}], "start": 1726477699021, "stop": 1726477699027}], "parameters": [{"name": "driver", "value": "<class 'selenium.webdriver.firefox.webdriver.WebDriver'>"}], "start": 1726477692725, "stop": 1726477699028, "uuid": "3654b4d4-b2b2-40ed-86ee-d838eed1412c", "historyId": "a5b0693b736fd1d4d9159f7e366b4ad7", "testCaseId": "1a6aa94b135b464a3c7527f211bb4bdb", "fullName": "tests.test_main_page.TestMainPage#test_making_order_by_authenticated_user_success", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_main_page"}, {"name": "subSuite", "value": "TestMainPage"}, {"name": "host", "value": "NB-ISC-69172.local"}, {"name": "thread", "value": "46944-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_main_page"}]}